lib/matplotlib/backend_tools.py

```diff
@@ -77,14 +77,35 @@ def __init__(self, toolmanager, name):
                       'experimental for now, the API will likely change in ' +
                       'version 2.1, and some tools might change name')
         self._name = name
+        self._toolmanager = toolmanager
         self._figure = None
-        self.toolmanager = toolmanager
-        self.figure = toolmanager.canvas.figure
 
     @property
     def figure(self):
         return self._figure
 
+    @figure.setter
+    def figure(self, figure):
+        self.set_figure(figure)
+
+    @property
+    def canvas(self):
+        return self.figure.canvas
```

Comment:
Rather than have the extra indirection via the property, this could be `self._figure.canvas`.